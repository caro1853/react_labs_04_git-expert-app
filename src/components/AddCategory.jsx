import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = (props) => {

    const [inputValue, setinputValue] = useState('');

    const onInputChange = (event)=> {
        setinputValue(event.target.value );
    }

    const onSubmit = (event)=> {
        event.preventDefault();
        if( inputValue.trim().length <= 1 ) return;
        console.log(inputValue);
        props.addNewCategory(inputValue);
        setinputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="add category">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            >
            </input>
        </form>
    );
}

AddCategory.propTypes = {
    addNewCategory: PropTypes.func.isRequired
}