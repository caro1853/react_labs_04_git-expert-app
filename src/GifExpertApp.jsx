import { useState } from "react"
import { AddCategory, GifGrid } from "./components";


export const GifExpertApp = () => {
    
    const [categories, setCategories ] = useState(['One Punch']);
    
    
    const addCategory = (newCategory)=>{
        setCategories( cate => {
            const cant = cate.filter(p=> p == newCategory).length;
            
            if(cant <= 0 ){
                return [...cate, newCategory];
            }
            else{
                return cate;
            }
        }
         );
    }


    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory addNewCategory = { addCategory }></AddCategory>

            { categories.map(cat => {
                return <GifGrid key = { cat } category = { cat }></GifGrid>
            })}
                

        </>
    )
}