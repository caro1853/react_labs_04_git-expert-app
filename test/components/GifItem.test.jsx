import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('GifItem', ()=> {
    const title = 'Dragon';
    const url = 'https://';
    
    test('Snapshot', ()=> {
        const { container } = render(<GifItem title={title} url={url}></GifItem>);
        expect( container ).toMatchSnapshot();
    });

    test('mostrar la imagen con el URL y ALT correcto', ()=> {
        render(<GifItem title={title} url={url}></GifItem>);
        //screen.debug();

        //console.log( screen.);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
        expect( screen.getByText(`${title}`) ).toBeTruthy()
    });
})