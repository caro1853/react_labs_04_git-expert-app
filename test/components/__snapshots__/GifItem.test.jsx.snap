// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon"
      src="https://"
    />
    <p>
      Dragon
    </p>
  </div>
</div>
`;
